main.go

```diff
@@ -116,7 +116,7 @@ func (s ReleaseSlice) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
 
-var versionRegex = regexp.MustCompile(`^(\d+\.\d+\.\d+)(_(\d{4}-\d{2}-\d{2}))?$`)
+var versionRegex = regexp.MustCompile(`^(\d+\.\d+\.\d+(?:-(0|[0-9A-Za-z-]*(?:\.(?:0|[0-9A-Za-z-]*))*))?)(?:_(\d{4}-\d{2}-\d{2}))?$`)
```

Comment:
I'm a bit overwhelmed by the complexity of that regexp. Isn't `0|[0-9A-Za-z-]*` and `[0-9A-Za-z-]*` equivalent?

That would for example allow simplifying `(?:\.(?:0|[0-9A-Za-z-]*))` to `(?:\.[0-9A-Za-z-]*)`.

The overall regexp can then be simplified to `^(\d+\.\d+\.\d+(?:-[0-9A-Za-z-]*(?:\.[0-9A-Za-z-]*)*)?)(?:_(\d{4}-\d{2}-\d{2}))?$`. As two consecutive dots in the version number are odd, it's probably better to change it to
`^(\d+\.\d+\.\d+(?:-[0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*)?)(?:_(\d{4}-\d{2}-\d{2}))?$`. I've also removed one unnecessary match group.

Reply:
Or we just simplify the first part to `(\d+\.\d+\.\d+(?:-[0-9A-Za-z.-]+)?)` and don't verify whether the version is actually semver.

Alternatively, we could just use https://pkg.go.dev/golang.org/x/mod/semver to parse the first part of the filename. That would also allow ensuring that calens maintains a consistent order if there are multiple releases on the same day.

Reply:
On second thought, let's just simplify the regexp to `^([^_]+)(?:_(\d{4}-\d{2}-\d{2}))?$` and use the semver library to verify the version. That's much better than using an overcomplicated regexp.

Reply:
ok, i simplified the regex and added the semver pkg to parse the version. It will let the script die when the version syntax is not semver compliant.